lib/octo_events_web/views/events_view.ex

```diff
@@ -15,4 +15,11 @@ defmodule OctoEventsWeb.EventsView do
       }
     }
   end
+
+  def render("show_events.json", %{events: events}) do
+    events
+    |> Enum.map(fn event ->
+      %{action: event.action, created_at: event.created_at}
+    end)
+  end
```

Comment:
**suggestion:** uma boa opção para manter a definição do payload que será retornado em uma única função seria utilizar as macros `render_many` e `render_one`, exemplo:
```elixir
def render("index.json", %{events: events}) do
    render_many(events, __MODULE__, "show.json")
  end

  def render("show.json", %{event: event}) do
    render_one(event, __MODULE__, "event.json")
  end

  def render("event.json", %{event: event}) do
    %{
      id: event.id,
      action: event.action,
      created_at: event.created_at
    }
  end
````

Reply:
Só esse cara que não consegui, me da um erro de falta de template

`
** (Phoenix.Template.UndefinedError) Could not render "event.json" for OctoEventsWeb.EventsView, please define a matching clause for render/2 or define a template at "lib/octo_events_web/templates/events/*". No templates were compiled for this module.
     Assigns:
     
     %{events: %OctoEvents.Event{__meta__: #Ecto.Schema.Metadata<:loaded, "events">, action: "closed", created_at: "2022-09-26T18:21:21Z", id: 406, number: 4}, view_module: OctoEventsWeb.EventsView, view_template: "event.json"}
     
     Assigned keys: [:events, :view_module, :view_template]
`

Reply:
Agora foi!